lib/common/ipc_controld.c

```diff
@@ -417,7 +420,10 @@ pcmk_controld_api_node_info(pcmk_ipc_api_t *api, uint32_t nodeid)
         return EINVAL;
     }
     if (nodeid > 0) {
-        pcmk__xe_set_id(request, "%lu", (unsigned long) nodeid);
+        char *nodeid_s = crm_strdup_printf(PRIu32, nodeid);
```

Comment:
This was a regression introduced in 2.0.5 by b9539da2 and ae14fa4a

Reply:
How do you figure? It wasn't a problem until we replaced `crm_xml_set_id()` with `pcmk__xe_set_id()` in 09d13ae7 (not in any release yet)

Incidentally there are two places in `crm_node.c` itself that mess up the ID this way

Reply:
Ah, I thought `crm_xml_set_id()` behaved the same way

Also lib/cluster/corosync.c and lib/common/ipc_client.c. We need to fix all of them for rc4

Reply:
Ignore that last bit, of course it's not in 2.1.8 :)